### PR TITLE
add list-strategies subcommand

### PR DESCRIPTION
**Pull request by xmatthias**
## Summary
Add list-strategies subcommand allowing to list all strategies in a folder.
Default output is a table, but machine-readable `--one-column` output is available as well.

## Quick changelog

- Refactor resolvers further to use class-mehods to reduce code duplication
- Combine `_load_*()` methods into a generalized one (apart from load_strategy due to loading from base64)
- Allow `IResolver._get_valid_object()` to work without a name, where it returns all valid objects within that file
- move module instanciation to `_load_object()` to support searching modules without initializing them.

### CODE REVIEW

File: docs/utils.md [hunk 1]
```diff
@@ -108,6 +108,47 @@ With custom user directory
...
+                        Path to userdata directory.
+```
+
+!!! Warning
+    Using this command will try to load all python files from a folder. This can be a security risk if untrusted files reside in this folder, since all module-level code is executed.
+
+Example: search default strategy folder within userdir
+
+``` bash
+freqtrade list-strategies --user-data ~/.freqtrade/
```
Comment (hroff-1902) [inline docs/utils.md]: "--userdir"

File: docs/utils.md [hunk 2]
```diff
@@ -108,6 +108,47 @@ With custom user directory
 freqtrade new-hyperopt --userdir ~/.freqtrade/ --hyperopt AwesomeHyperopt
 ```
 
+## List Strategies
+
+Use the `list-strategies` subcommand to see all strategies in one particular folder.
```
Comment (hroff-1902) [inline docs/utils.md]: no folders 😉 
(pls also check other appearances)

File: docs/utils.md (hunk 1 above)
Comment (xmatthias) [reply docs/utils.md]: `--user-data-dir` works as well ... which shortcuts to `--user-data` ... but i agree, we should be consistent :D

**Comment by xmatthias** [comment]
> I propose to print only filename (like sample_strategy.py) in the location column instead of full path. It's redundant, we are listing a single dir...

good point - done.

**Pull request merged**